src/runtime/HalideBuffer.h

```diff
@@ -1045,10 +1045,11 @@ class Buffer {
 
     /** Return a typed reference to this Buffer. Useful for converting
      * a reference to a Buffer<void> to a reference to, for example, a
-     * Buffer<const uint8_t>. Does a runtime assert if the source
-     * buffer type is void. */
+     * Buffer<const uint8_t>, or converting a Buffer<T>& to Buffer<const T>&.
```

Comment:
Oops, my confusion wasn't about ampersand placement - I misread the combination of new and deleted lines to say that it was illegal to cast a T to a const T.